data/src/main/java/com/microsoft/azure/kusto/data/format/CslFormat.java

```diff
@@ -0,0 +1,38 @@
+package com.microsoft.azure.kusto.data.format;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public abstract class CslFormat {
+    public abstract String getType();
+
+    public abstract Object getValue();
+
+    // Value must not be null
+    abstract String getValueAsString();
+
+    private String getValueOrNullAsString() {
+        if (getValue() == null) {
+            return "null";
+        } else {
+            return getValueAsString();
+        }
+    }
+
+    public String toString() {
+        return getType() + "(" + getValueOrNullAsString() + ")";
+    }
+
+    // For example, parses "int(7)" as "7"
+    // If type wasn't found in valueWithType, returns it unchanged
+    public static String parseValueFromValueWithType(String valueWithType, String type) {
+        String valueWithTypeRegex = String.format("%s\\s*\\(\\s*(.*\\S+)\\s*\\)\\s*", type);

```

Comment:
I'm pretty sure the ".*" is not needed

Reply:
We need it in case there's a whitespace character in the matching group.

Reply:
I tried it in regex101, it's the same results with or without it.
Do you want to capture the whitespace inside the parens or not?

Reply:
I haven't tried a tester, but I indeed want to capture any whitespace that's part of the value inside the parens.

Reply:
That's part of the value = includes leading and trailing?
Those are your options (unescaped regexes):

```regex
int\s*\((.*)\)\s* 

int(aaaa) => "aaaa"
int( aaaa  )  => " aaaa  "
int(a a) => "a a"
int( a a ) => " a a "

int\s*\(\s*(.*?)\s*\)\s*

int(aaaa) => "aaaa"
int( aaaa  )  => "aaaa"
int(a a) => "a a"
int( a a ) => "a a"

```

Reply:
It doesn't make much sense for `int`, but consider others, like `datetime(2022-01-05 9:59:32)`

Reply:
Ok I see what you mean, but I still prefer `\s*\(\s*(.*?)\s*\)\s*` (it's equivalent)

Reply:
Are you sure they're equivalent? I think mine enforces that there's some non-whitespace inside the capture group, and I don't think yours does.

Reply:
Ok yes if that's something we want we can keep it as is